python/GafferImageUI/ImageViewUI.py

```diff
@@ -1260,10 +1260,11 @@ def __setValue( self, value, *unused ) :
 		self.getPlug().setValue( value )
 
 		with self.getContext():
-			inputImage = self.getPlug().node()._getPreprocessor()["_selectView"]["out"]
+			inputImage = self.getPlug().node()._getPreprocessor()["_selectView"]["in"]
 			viewNames = inputImage.viewNames()
 
 		if turningOn and len( viewNames ) > 1:
+			node = self.getPlug().node()
 			if node["compare"]["view"].getValue() == node["view"].getValue() or not node["compare"]["view"].getValue() in viewNames :
 				# We're turning on comparison, but currently the comparison view is identical to the
 				# default view - pick a different view, so we actually get a meaningful comparison
```

Comment:
Based on this comment, it seems like all this logic is obsolete now anyway. We're not using different views for comparison, and there's not even a UI for the user to choose a view for comparison. So I think we want the comparison view to always be the same as the main view don't we?